### ISSUE TITLE
Support addons that apply to both the website and editor

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#6409]
I don't think we've ever made such an addon.

#6400 is currently stuck here, having had "Validate addon manifest" [fail](https://github.com/ScratchAddons/ScratchAddons/actions/runs/5580747470/jobs/10198066396).

And I tried including both the "community" and "editor" tags and for one, the addon has both a "for website" label and a "for editor" label when filtering by themes in the settings page, and second, it doesn't show up in Themes › Editor themes, only in Themes › Website themes.

**Comment by DNin01** [comment]
Pinging @ScratchAddons/admins - I think you'll know what to do.

**Comment by WorldLanguages** [comment]
Why not split the addon in two?

**Comment by BroJac5246** [comment]
> Why not split the addon in two?

Because the two would do the same thing so ideally they'd be the same addon.

**Comment by DNin01** [comment]
> Why not split the addon in two?

Technically, if there was only one feature from that addon that applied to a certain area (like the editor), then it could be separated, leaving us with an addon that only applies to the website or only applies to the editor.

Or we could do the same thing but move the feature to a customizable-colors addon.

Or maybe we can simply remove the "high contrast switches" setting for now so we don't have to worry about this.

**Comment by WorldLanguages** [comment]
> Because the two would do the same thing so ideally they'd be the same addon.

In general, it's pretty hard to do *the exact same thing* in both the website and the editor. What specific feature is this about?

**Comment by WorldLanguages** [comment]
Addons with a common purpose can be linked together by the "related addons" feature (#5937)

It's probably still a good idea to keep editor addons separate, for various reasons, including simpler addon support by Scratch forks (as those wouldn't need to hide settings that belong to the website nor ignore scripts/styles that only apply to the website)

**Comment by BroJac5246** [comment]
> Addons with a common purpose can be linked together by the "related addons" feature (#5937)

This is still not ideal, however.

> It's probably still a good idea to keep editor addons separate, for various reasons, including simpler addon support by Scratch forks (as those wouldn't need to hide settings that belong to the website nor ignore scripts/styles that only apply to the website)

Could we have a way to link a website addon and an editor addon so that only one is shown but both are toggled at once? They should share settings and to the end-user they would effectively be one addon, but this would make them easier to separate in Scratch forks.

**Comment by WorldLanguages** [comment]
> Could we have a way to link a website addon and an editor addon so that only one is shown but both are toggled at once? They should share settings and to the end-user they would effectively be one addon, but this would make them easier to separate in Scratch forks.

Yes, we could explore ideas like that in the future. I've also seen situations where it would be nice to allow addons to include other addon's settings in their settings page card. Also a possibility to have boolean settings that toggle another addon.
